src/internal/connector/exchange/container_resolver.go

```diff
@@ -86,8 +86,16 @@ func (cr *containerResolver) PathInCache(pathString string) (string, bool) {
 // addFolder adds a folder to the cache with the given ID. If the item is
 // already in the cache does nothing. The path for the item is not modified.
 func (cr *containerResolver) addFolder(cf graph.CacheFolder) error {
-	// Only require a non-nil non-empty parent if the path isn't already
-	// populated.
+	// deleted folders bypass the other checks
+	addtl := cf.GetAdditionalData()
+	_, cf.Removed = addtl[graph.AddtlDataRemoved]
+
+	if cf.Removed {
+		cr.cache[*cf.GetId()] = &cf
+		return nil
+	}
+
```

Comment:
Makes sense.  I was imagining this as running the delta token from root, whereupon we'd see the @removed additional data.  But since we don't re-use that link, this change is moot.